pkg/importer/avro/avro_util.arrai

```diff
@@ -0,0 +1,143 @@
+    cond transformPrimitiveType(name) {
+        'NonPrimitiveType': //str.title(name),
+        _: name
+    }
+;
+
+# Get type name from types which looks like:
+# ['null', 'string'], ['string', 'int'] or ['null', {'type':'record', ...}] etc.
+let getTypeName = \t
+    cond t {
+        (:s): (s: correctName(s)),
+        {'type': (s: type), 'items': (:s), ...}:
+            cond type {
+                'map': (s: $'${correctName(t('items').s)}Set', type_S: $'set of ${correctName(t('items').s)}'), # will be displayed in union definition
+                'array': (s: $'${correctName(t('items').s)}Sequence', type_S: $'sequence of ${correctName(t('items').s)}'), # will be displayed in union definition
+            },
+        {'type': (s: type), 'items': {'type': (:s), ...}, ...}:
+            cond type {
+                'map': (s: $'${correctName(t('items')('name').s)}Set', type_S: $'set of ${correctName(t('items')('name').s)}'), # will be displayed in union definition
+                'array': (s: $'${correctName(t('items')('name').s)}Sequence', type_S: $'sequence of ${correctName(t('items')('name').s)}'), # will be displayed in union definition
+            },
+        {'type': (:s), ...}: (s : correctName(t('name').s)),
+    }
+;
+
+let combineTypes = \types \handler
+    let fullTypeNames = types >> getTypeName(.);
+    let types = fullTypeNames >> .s;
+    cond //seq.contains(['null'], types) {
+        true:
+            let typeNames = fullTypeNames where .@item.s != 'null' rank (:.@);
+            cond types count {
+                2: //seq.concat(typeNames >> handler(.)) + '?',
+                _: //seq.concat(typeNames >> //str.title(.s) orderby .@item >> .@item) + '?'
+            }
+        ,
+        _: //seq.concat(types >> //str.title(.) orderby .@item >> .@item)
+    }
+;
+
+let transformType = \type
+    cond type {
+        # (s: 'string') etc.
+        (s: typeName):
+            let correctedName = correctName(typeName);
+            cond correctedName {
+                "null": "string[~null]",
+                _: correctedName,
+            },
+        (a: typeArray):
+            # type can be ['null', 'string'], ['string', 'int'] or ['null', {'type':'record', ...}] etc.
+            # [(s: 'null'), (s: 'string')] to ['null', 'string']
+            combineTypes(typeArray, \. cond . {(type_S:type_S, ...): type_S, _: .s}),
+        {'logicalType': logicalTypeName, 'type': typeName, ...}: # Must be before {'type': typeName, ...}
+            transformLogicalTypes(type), 
+        {'type': typeName, ...}:
+            # it is array, map
+            cond typeName.s {
+                'array': 'sequence of ' + cond type('items') {{'name': (:s), ...}: correctName(s), (:s): correctName(s),},
+                'map': 'set of ' + cond type('items') {{'name': (:s), ...}: 'String' + correctName(s) + 'Item', (:s): correctName(s),},
+                _: correctName(type('name').s)
+            },
+    }
+;
+
+let getDefaultVal = \.
```

Comment:
Nit: Avoid verbs in function names for simple transformations. This would read better as just `defaultVal`, since that's what it returns. What it does (i.e. `get`) is not interesting in the context of the calling code.

Also avoid using `\.` in most cases. It works, but you're passing up an opportunity to document the data you're working with. What kind of input is this function expecting? kinds of thing have a "defaultVal"? I have to be familiar with Avro to know that, and I shouldn't need to know that to understand your code.
```suggestion
let defaultVal = \.
```